### PR TITLE
perf(nuxt): mark payload as shallowReactive

### LINKED ISSUE #26825: move `nuxtApp.payload` to `shallowReactive` and move `renderJsonPayloads` out of experimental
### Describe the feature

Context:
https://github.com/nuxt/nuxt/pull/20218
https://github.com/nuxt/nuxt/pull/19205

`renderJsonPayloads` has been enabled by default since 3.5 has hasn't changed since. We probably can move this out of experimental and apply shallowReactive to the payload instead of marking it as reactive

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/26825

### ❓ Type of change
- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This improves performance by not marking the entire Nuxt payload as reactive - but granularly enabling shallow reactivity first in the parts of it that we need to be reactive.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/app/nuxt.ts [hunk 1]
```diff
@@ -252,12 +252,12 @@ export function createNuxtApp (options: CreateOptions) {
...
-      data: {},
-      state: {},
+    payload: shallowReactive({
+      data: shallowReactive({}),
+      state: reactive({}),
       once: new Set<string>(),
-      _errors: {},
-      ...(import.meta.client ? window.__NUXT__ ?? {} : { serverRendered: true }),
+      _errors: shallowReactive({}),
+      error: undefined,
```
Comment (huang-julien) [inline packages/nuxt/src/app/nuxt.ts:260]:
```suggestion
```
Is it needed ? This is not going to be serialized if undefined

Comment (danielroe) [reply packages/nuxt/src/app/nuxt.ts]: Good spot. I think I added it at some point to ensure that `useError` continued to work - but that was when I was experimenting with not even marking the payload as `shallowReactive`.

https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/app/composables/error.ts#L10

**Comment by daniluk4000** [comment]
That's sick 🔥

**Comment by Mini-ghost** [comment]
@danielroe

Since this PR marks `payload` as `shallowReactive`, should the `reactive` that was removed in this PR #26558 be restored? Or, as mentioned at the PR end, does `runtimeConfig` not require reactivity?

**Comment by danielroe** [comment]
I think it might not require reactivity at all.

**Pull request merged**